src/solace_ai_connector/components/general/websearch/websearch_base.py

```diff
@@ -0,0 +1,44 @@
+"""Base class for Web Search"""
+from ...component_base import ComponentBase
+
+info_base = {
+    "class_name": "WebSearchBase",
+    "description": "Base class for performing a query on web search engines.",
+    "config_parameters": [
+        {
+            "name": "engine",
+            "required": True,
+            "description": "The type of search engine.",
+            "default": "DuckDuckGo"
+        },
+        {
+            "name": "detail",
+            "required": False,
+            "description": "Return the detail.",
+            "default": False
+        }
+    ],
+    "input_schema": {
+        "type": "object",
+        "properties": {},
+    },
+    "output_schema": {
+        "type": "object",
+        "properties": {},
+    }
+}
+
+class WebSearchBase(ComponentBase):
+    def __init__(self, info_base, **kwargs):
+        super().__init__(info_base, **kwargs)
+        self.engine = self.get_config("engine")
+        self.detail = self.get_config("detail")
```

Comment:
This component can be updated to be a generic component that can load all, see my earlier comment